worldedit-core/src/main/java/com/sk89q/worldedit/command/SchematicCommands.java

```diff
@@ -470,9 +470,18 @@ public void save(
             }
         }
 
-        //FAWE end
         File f = worldEdit.getSafeSaveFile(actor, dir, filename, format.getPrimaryFileExtension());
-
+        int i = f.getName().lastIndexOf('.');
+        if (i == -1 && f.getName().isEmpty() || i == 0) {
```

Comment:
Can a file name be emtpy?
We're checking both `i == - 1` and the empty name two times, maybe it can be extracted into variables for readability.